packages/app/src/app/overmind/effects/vscode/index.ts

```diff
@@ -496,6 +496,29 @@ export class VSCodeEffect {
     }
   };
 
+  public async openDiff(sandboxId: string, module: Module, oldCode: string) {
+    const recoverPath = `/recover/${sandboxId}/recover-${module.title}`;
+    const filePath = `/sandbox${module.path}`;
+    const fileSystem = window.BrowserFS.BFSRequire('fs');
+
+    if (!fileSystem.existsSync(`/recover/${sandboxId}`)) {
+      fileSystem.mkdirSync(`/recover/${sandboxId}`);
+    }
+    fileSystem.writeFileSync(recoverPath, oldCode);
+
+    this.editorApi.editorService.openEditor({
```

Comment:
Let's add a comment on what the heck is happening here 😂. This is some really tricky logic, we're dancing around VSCode here.